File: android-working-old/app/src/main/java/uk/co/darkerwaters/scorepal/score/pingpong/PingPongPoint.java
package uk.co.darkerwaters.scorepal.score.pingpong;

import android.content.Context;

import uk.co.darkerwaters.scorepal.R;
import uk.co.darkerwaters.scorepal.score.base.Point;

public enum PingPongPoint implements Point {
    ZERO(0, R.string.display_zero, R.string.speak_zero, R.string.speak_zeros),
    ROUND(-1, R.string.display_round, R.string.speak_round, R.string.speak_rounds),
    POINT(-2, R.string.points, R.string.speak_point, R.string.speak_points),
    MATCH(-3, R.string.display_match, R.string.speak_match, R.string.speak_match),
    DEUCE(-4, R.string.display_deuce, R.string.speak_deuce, R.string.speak_deuce);

    private final int value;
    private final int displayStrId;
    private final int speakStrId;
    private final int speakStrPluralId;

    PingPongPoint(int value, int displayStrId, int speakStrId, int speakStrPluralId) {
        this.value = value;
        this.displayStrId = displayStrId;
        this.speakStrId = speakStrId;
        this.speakStrPluralId = speakStrPluralId;
    }
    @Override
    public int val() {
        return this.value;
    }
    @Override
    public String displayString(Context context) {
        if (null != context && 0 != this.displayStrId) {
            return context.getString(this.displayStrId);
        }
        else {
            return Integer.toString(this.value);
        }
    }
    @Override
    public String speakString(Context context) {
        if (null != context && 0 != this.speakStrId) {
            return context.getString(this.speakStrId);
        }
        else {
            return Integer.toString(this.value);
        }
    }
    @Override
    public String speakString(Context context, int number) {
        if (null != context) {
            if (number == 1 && 0 != this.speakStrId) {
                // in the singular
                return context.getString(this.speakStrId);
            }
            else if (0 != this.speakStrPluralId) {
                return context.getString(this.speakStrPluralId);
            }
            else {
                return Integer.toString(this.value);
            }
        }
        else {
            return Integer.toString(this.value);
        }
    }
    @Override
    public String speakAllString(Context context) {
        // just say the number then 'all'
        return speakString(context) + " " + context.getString(R.string.speak_all);
    }
}

File: multiphone/lib/match/points/ping_pong_point.dart
import 'package:flutter/cupertino.dart';
import 'package:multiphone/helpers/values.dart';
import 'package:multiphone/match/points/point.dart';

class PingPongPoint implements Point {
  static final zero = PingPongPoint(
    0,
    (context) => Values(context).strings.display_zero,
    (context) => Values(context).strings.speak_zero,
    (context) => Values(context).strings.speak_zeros,
  );
  static final round = PingPongPoint(
    -1,
    (context) => Values(context).strings.display_round,
    (context) => Values(context).strings.speak_round,
    (context) => Values(context).strings.speak_rounds,
  );
  static final point = PingPongPoint(
    -2,
    (context) => Values(context).strings.points,
    (context) => Values(context).strings.speak_point,
    (context) => Values(context).strings.speak_points,
  );
  static final match = PingPongPoint(
    -3,
    (context) => Values(context).strings.display_match,
    (context) => Values(context).strings.speak_match,
    (context) => Values(context).strings.speak_match,
  );
  static final deuce = PingPongPoint(
    -4,
    (context) => Values(context).strings.display_deuce,
    (context) => Values(context).strings.speak_deuce,
    (context) => Values(context).strings.speak_deuce,
  );

  final int value;
  final String Function(BuildContext) displayStrId;
  final String Function(BuildContext) speakStrId;
  final String Function(BuildContext) speakStrPluralId;

  PingPongPoint(
    this.value,
    this.displayStrId,
    this.speakStrId,
    this.speakStrPluralId,
  );

  @override
  int val() {
    return this.value;
  }

  @override
  String displayString(BuildContext context) {
    if (null != context && null != this.displayStrId) {
      return this.displayStrId(context);
    } else {
      return this.value.toString();
    }
  }

  @override
  String speakString(BuildContext context) {
    if (null != context && null != this.speakStrId) {
      return this.speakStrId(context);
    } else {
      return this.value.toString();
    }
  }

  @override
  String speakNumberString(BuildContext context, int number) {
    if (null != context) {
      if (number == 1 && null != this.speakStrId) {
        // in the singular
        return this.speakStrId(context);
      } else if (null != this.speakStrPluralId) {
        return this.speakStrPluralId(context);
      } else {
        return this.value.toString();
      }
    } else {
      return this.value.toString();
    }
  }

  @override
  String speakAllString(BuildContext context) {
    // just say the number then 'all'
    return '${speakString(context)} ${Values(context).strings.speak_all}';
  }
}
